src/NLog/Internal/FileAppenders/BaseFileAppender.cs

```diff
@@ -337,24 +337,19 @@ private FileStream TryCreateFileStream(bool allowFileSharedWriting)
 
         private void UpdateCreationTime()
         {
-            if (File.Exists(this.FileName))
+            FileInfo fileInfo = new FileInfo(this.FileName);
+            if (fileInfo.Exists)
             {
-#if !SILVERLIGHT
-                this.CreationTimeUtc = File.GetCreationTimeUtc(this.FileName);
-#else
-                this.CreationTimeUtc = File.GetCreationTime(this.FileName);
-#endif
+                this.CreationTimeUtc = FileCharacteristicsHelper.ValidateFileCreationTime(fileInfo, (f) => f.GetCreationTimeUtc(), (f) => f.GetLastWriteTimeUtc()).Value;
```

Comment:
This code is called when opening the file. So it is only when using the RetryingMultiProcessAppender it is called a lot.

Also the lambdas are completely static without capture of this, so the compiler should turn them into zero allocations.

Ran a performance profiler of memory allocations using the RetryingMultiProcessAppender, and it within the BaseFileAppender.UpdateCreationTime() then the most expensive was the FileInfo-allocation, and the second most expensive was assigning the CreationTimeUtc where it used DateTime.ToLocalTime().